src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -33,47 +33,72 @@ public JsonSchemaTransformer(JsonSchema schema)
         public (List<Error> errors, JToken token) TransformContent(Document file, Context context, JToken token)
         {
             var uidCount = _uidCountCache.GetOrAdd(file, GetFileUidCount(_schema, token));
-            return TransformContentCore(file, context, _schema, token, uidCount);
+            return TransformContentCore(file, _schema, token, uidCount, context.MarkdownEngine, context.LinkResolver, context.XrefResolver);
         }
 
-        public (List<Error>, IReadOnlyList<InternalXrefSpec>) LoadXrefSpecs(Document file, Context context, JToken token)
+        public (List<Error>, IReadOnlyList<InternalXrefSpec>) LoadXrefSpecs(
+            Document file,
+            JToken token,
+            MarkdownEngine markdownEngine,
```

Comment:
Move these services to the constructor of `JsonSchemaTransformer`?

Reply:
I could give it a try, ideally we should not pass around the component instances, but `JsonSchemaTransformer` is not hold in `context`.